src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorResolver.cs

```diff
@@ -67,8 +74,7 @@ public IEnumerable<TagHelperDescriptor> Resolve([NotNull] TagHelperDescriptorRes
                 }
                 catch (Exception ex)
                 {
-                    var directiveName = "@" + directiveDescriptor.DirectiveType.ToString().ToLowerInvariant();
-
+                    var directiveName = "@" + _directiveNames[directiveDescriptor.DirectiveType];
```

Comment:
:-1: this class _must_ change if we ever add another directive that reaches here. but I'll add a debug assertion that the dictionary contains the value.